File: backend/src/main/java/fr/irvineflth/EcoFleet/service/VehicleService.java
package fr.irvineflth.EcoFleet.service;

import fr.irvineflth.EcoFleet.dto.VehicleDto;

import java.util.List;

public interface VehicleService {
    List<VehicleDto> findAll();
    VehicleDto findById(long id);
    VehicleDto create(VehicleDto vehicleDto);
    VehicleDto update(long id, VehicleDto vehicleDto);
    void delete(long id);
}

File: frontend/src/app/core/services/vehicle.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Vehicle} from '../models/vehicle.model';
import {VehicleDto} from '../dto/vehicle.dto';

@Injectable({
  providedIn: 'root'
})
export class VehicleService {
  private baseUrl = 'http://localhost:8080/api/vehicles';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Vehicle[]> {
    return this.http.get<Vehicle[]>(this.baseUrl);
  }

  getById(id: number): Observable<Vehicle> {
    return this.http.get<Vehicle>(`${this.baseUrl}/${id}`);
  }

  create(dto: VehicleDto): Observable<Vehicle> {
    return this.http.post<Vehicle>(this.baseUrl, dto);
  }

  update(id: number, dto: VehicleDto): Observable<Vehicle> {
    return this.http.put<Vehicle>(`${this.baseUrl}/${id}`, dto);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
